### PR TITLE
modify building CDS object from 10x-type data

### PR DESCRIPTION
**Pull request by a-solovyev12**
- the current version of the package (0.0.5) breaks when attempting to parse data from 10x genomics directory
- this PR modifies parsing the raw data to account for different input formats 
- we also reduce repeated code, and account for optional presence of row/column labels in metadata files.

### CODE REVIEW

File: R/cmd_methods.R [hunk 1]
```diff
@@ -113,26 +113,69 @@ monocle_create <- function(
...
+            sep = "\t"
+        else sep = ","
+
+        # Read file once, and adjust row/columns as appropriate before
+        # returning as matrix
+
+        if ( file_type == 'expression_matrix' ){
+            mat = as.matrix(read.delim(file, sep = sep, stringsAsFactors = FALSE, row.names = 1)) 
+        }else{
+            mat = read.delim(file, sep = sep, stringsAsFactors = FALSE, row.names = NULL)
```
Comment (nh3) [inline R/cmd_methods.R]: `read.delim` by default treat first row as header, so for 10x files where there's no header, the check later for row counts will not work.

File: R/cmd_methods.R [hunk 2]
```diff
@@ -113,26 +113,69 @@ monocle_create <- function(
...
+            if (nrows(mat) != expected_rows){
+                if (nrows(mat) - 1 == expected_rows){
+                    mat <- mat[-1,]
+                }else{
+                    message(paste('Number of', file_type, 'entries not appropriate to matrix.'))
+                    q(save = 'no', status = 1)
+                }
+            }
+
+            # Move first column into rownames where possible
```
Comment (nh3) [inline R/cmd_methods.R]: `new_cell_data_set()` seems to be able to handle empty metadata table as long as the row names exist, and `as.matrix()` keeps rownames, so `as.matrix(..., read.delim(row.names=1))` works fine. But the approach here is also ok.

File: R/cmd_methods.R (hunk 1 above)
Comment (pinin4fjords) [reply R/cmd_methods.R]: Oh yeah- forgot my header = F!

File: R/cmd_methods.R (hunk 2 above)
Comment (pinin4fjords) [reply R/cmd_methods.R]: Ahh, I hadn't tested that, but was told it wouldn't work - now simplified.

**Comment by ktpolanski** [comment]
Hey guys, I took a look at your pull request and would feel more comfortable proceeding with my existing code base. Hopefully that's okay, sorry about the lack of communication earlier. There are still some minor things in your implementation (e.g. dropping headers for metadata text imports) that would need tweaking, but I will mirror your superior dimensional compatibility logic in my version.

**Comment by pinin4fjords** [comment]
Great, thank you for the response. Look forward to testing the fix soon.

**Comment by a-solovyev12** [comment]
@ktpolanski No problem, but could you avoid merging your PR before we run our tests on the fix you are going to make? Need to make sure the script does not break when parsing 10x-type data, which is an important use case in one of our projects.

**Comment by ktpolanski** [comment]
All the merging power is in nh3's hands anyway, but feel free to do any testing you may need. I've added a 10x-style mtx+tsv combo to the tests, and also create dummy names for gene annotations.

**Comment by pinin4fjords** [comment]
Closing in favour of https://github.com/ebi-gene-expression-group/monocle-scripts/pull/8

**Pull request closed without merging**